File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n = nums.size();

        unordered_set<int> set;

        for(auto& num:nums){
            set.insert(num);
        }

        int maxCnt = 0;
        
        for(auto& num:set){
            if(set.find(num-1)==set.end()){
                int cnt = 1;
                int x = num;
                while(set.find(x+1)!=set.end()){
                    cnt++;
                    x++;
                }
                maxCnt = max(maxCnt, cnt);
            }
        }
        return maxCnt;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        hashset = set(nums)

        count = 0
        maxcount = 0

        for num in hashset:
            if num-1 in hashset:
                # this is not the starting of the consequetive seq
                continue
            
            else:
                temp = num
                count = 1
                while temp + 1 in hashset:
                    count += 1
                    temp = temp + 1
                
                maxcount = max(maxcount,count)

        return maxcount 